### PR TITLE
New layout

### PR DESCRIPTION
**Pull request by sri2k1us**
Review Visual Diffs here - https://www.chromatic.com/pullrequest?appId=5ebad5c6b991d60022b1e72e&number=105

- Laptop
image

- iPhone 5 / SE 
image

- iPad (50 %)
image

- Data - large folder names with new layout
image

- Data iPhone 5/ SE - Folder selection instead of bread crumbs
image

- Apps
image

- Apps - iPhone 5 / SE 
image

- Analyses
image

- Batch drill down
image

- Analyses - iPhone 5 / SE
image

### CODE REVIEW

**Comment by sri2k1us** [comment]
Updating to WIP. I have started making more changes to reflect the design we finalized on 05/05/2020 UX meeting.

File: src/components/data/CreateFolderDialog.js [hunk 1]
```diff
@@ -92,7 +92,7 @@ function CreateFolderDialog(props) {
                         <Dialog
                             open={open}
                             onClose={onClose}
-                            maxWidth="xs"
+                            maxWidth="sm"
```
Comment (sri2k1us) [inline src/components/data/CreateFolderDialog.js:95]: `xs` was cutting off large folder names / deep paths.

File: src/components/data/toolbar/DataDotMenu.js [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+                onFolderCreated={() => {
+                    onCreateFolderDlgClose();
+                    refreshListing();
+                }}
+            />
+        </>
+    );
+}
+
+export default withI18N(DataDotMenu, messages);
```
Comment (sri2k1us) [inline src/components/data/toolbar/DataDotMenu.js:138]: I have removed menuItems that have not been implemented yet to help testing / clean up.

**Comment by sri2k1us** [comment]
This PR is now ready for review.

**Comment by sri2k1us** [comment]
**New Workflow to review PRs:** 
- First go to https://github.com/settings/tokens and create a Personal Access Token and name it chromatic (required only once)
- Next go to https://github.com/<username>/sonora/settings/secrets and create 2 secrets. 
1 for your personal token (created in first step) and next one for chromatic project token which can be found in Chromatic settings page.  (required only once).
- As usual create a branch in your fork and continue working
- I have set up chromatic  CI to run on every push. This is will build and deploy storybook. Note this job will run only on your fork and NOT in Sonora master (unless you push directly to master )
- Once you create the PR, copy the link spewed by the Chormatic CI ( like the one I pasted above) and put that in your PR description)
- Anyone can review the visual diffs, approve /request changes. Note this is in addition to code reviews in github.
Rejoice :tada:

File: src/components/apps/toolbar/Toolbar.js [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+                        startIcon={<Info />}
+                    >
+                        {getMessage("details")}
+                    </Button>
+                )}
+            </Hidden>
+            <IconButton className={classes.toolbarItems}>
+                <MoreVertIcon />
+            </IconButton>
+        </Toolbar>
```
Comment (sri2k1us) [inline src/components/apps/toolbar/Toolbar.js:131]: Placeholder for Dot Menu.

File: src/components/analyses/AnalysesNavigation.js [hunk 4]
```diff
@@ -190,6 +190,12 @@ function AnalysesNavigation(props) {
                     />
                 )}
             />
+            <Hidden xsDown>
+                <div className={classes.divider} />
+            </Hidden>
+            <IconButton>
+                <MoreVertIcon />
+            </IconButton>
```
Comment (sri2k1us) [inline src/components/analyses/AnalysesNavigation.js:198]: Placeholder for DotMenu.

File: src/components/data/utils.js [hunk 5]
```diff
@@ -66,10 +66,21 @@ const isReadable = (permission) => {
     );
 };
 
+const isOwner = (selectedResources) => {
+    if (!selectedResources) {
+        return false;
+    }
+    const notOwners = selectedResources.filter(
+        (resource) => resource.permission !== Permissions.OWN
```
Comment (ash3rz) [inline src/components/data/utils.js:74]: This should reuse `hasOwn`

File: src/server/index.js [hunk 6]
```diff
@@ -64,7 +64,7 @@ app.prepare()
         logger.info("Set up notification queue and websocket");
         setUpAmqpForNotifications();
         expressWs(server);
-        server.ws(NavigationConstants.NOTIFICATION_WS, function(ws, request) {
+        server.ws(NavigationConstants.NOTIFICATION_WS, function (ws, request) {
```
Comment (ash3rz) [inline src/server/index.js:67]: Is this from after running `prettier`?  I personally like the other look better 🤔

File: src/components/apps/toolbar/Toolbar.js [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+                renderInput={(params) => (
+                    <TextField
+                        {...params}
+                        label={formatMessage(intl, "filterLbl")}
+                        variant="outlined"
+                    />
+                )}
+            />
+            <div className={classes.divider} />
+            <Hidden smDown>
```
[the remark this answers is not in the document]
Comment (sri2k1us) [reply src/components/apps/toolbar/Toolbar.js:113]: Yes. In small screen, we decided to put everything under DotMenu( See Data DotMenu for reference) in our last UX meeting. Of course, Apps Dot menu is not done yet.

File: src/server/index.js (hunk 6 above)
Comment (sri2k1us) [reply src/server/index.js:67]: Yes. I updated `prettier` from `1.x` to `2.x`. May be we could look into configuring `prettier` to format it the way it did in `1.x`

**Comment by sri2k1us** [comment]
> LGTM 🚢

@ash3rz Did you manage to review in Chromatic as well ?

File: src/server/index.js (hunk 6 above)
Comment (psarando) [reply src/server/index.js:67]: I'm OK with the new default.
Here's a blurb about it: https://prettier.io/blog/2020/03/21/2.0.0.html#other-changes
Note that this is only for anonymous functions like this, and not something named like `function wsHandler(ws, request) { ... }`.

File: src/components/data/toolbar/UploadMenuItems.js [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+                </ListItemIcon>
+                <ListItemText primary={getMessage("browseLocal")} />
+            </MenuItem>
+        </label>,
+        <MenuItem
+            id={build(uploadMenuId, ids.IMPORT_MI)}
+            key={build(uploadMenuId, ids.IMPORT_MI)}
+            onClick={() => {
+                onImportFromURL();
+                console.log("Import from URL");
```
Comment (psarando) [inline src/components/data/toolbar/UploadMenuItems.js:47]: Can this debug log be removed now?

File: src/components/layout/CyVerseAppBar.js [hunk 9]
```diff
@@ -119,292 +234,336 @@ function CyverseAppBar(props) {
...
+                variant="outlined"
+                ref={ref}
+                className={clsx(classes.appBar, {
+                    [classes.appBarShift]: open,
+                })}
+            >
+                <Toolbar>
+                    <Hidden only={["sm", "md", "lg", "xl"]}>
+                        <IconButton
+                            aria-label="open drawer"
```
Comment (psarando) [inline src/components/layout/CyVerseAppBar.js:450]: Needs i18n.

File: src/components/data/Header.js [hunk 10]
```diff
@@ -1,163 +0,0 @@
...
-                    id={build(headerId, ids.SHARE_BTN)}
-                    variant="contained"
-                    disableElevation
-                    color="primary"
-                    className={classes.button}
-                    onClick={() => console.log("Share")}
-                >
-                    <Share className={classes.buttonIcon} />
-                    <Hidden xsDown>{getMessage("share")}</Hidden>
-                </Button>
```
[the remark this answers is not in the document]
Comment (sri2k1us) [reply src/components/data/Header.js:140]: Yes. I removed some buttons that are yet to be implemented. There was so much noice during implementation and rebase with master.

**Comment by sri2k1us** [comment]
Thanks for reviews. All pending PRs and new PRs will start chromatic. I will update instructions for chromatic in the README file.

**Pull request merged**